Client to server integration

OmerMunkMoveo opened the pull request.

OmerMunkMoveo committed:
started chef and restaurant communication 😎

OmerMunkMoveo committed:
all data connected to back

OmerMunkMoveo committed:
Implemented header search and auto complete. 🌭
next working on:
1. on click of search item: display the item.
2. styling the dropdown
3. debounce on the search
4. loaders
5. toastifiers

GuyLeiferMoveo commented on src/Components/Dishes/Dish.tsx:
```diff
@@ -8,15 +8,15 @@ import {Dish as DishModel} from "../../Interfaces/Dish"
 
 const Dish = (props: any) => {
-
+    // console.log(props.data.properties.split(','))
```
remove unnecessary comments

OmerMunkMoveo replied:
done

GuyLeiferMoveo commented on src/Components/Dishes/Dish.tsx:
```diff
@@ -40,13 +40,12 @@ const Dish = (props: any) => {
                 <div className="dish_props">
-                    {Object.keys(dish.properties).map((keyName:any, i: number) => {
-                        console.log(keyName, i)
+                    {dish.properties ? dish.properties.trim().split(',').map((property, index) => {
```
if the condition is only if, and not if else, use && and not ? : .

OmerMunkMoveo replied:
Done

GuyLeiferMoveo commented on src/Components/Dishes/Dishes.tsx:
```diff
@@ -3,19 +3,36 @@ import Dish from "./Dish";
+            const result = res.data.data
+            console.log(result)
+            const rendered: typeof Dish[] | JSX.Element[]= result.map((dish: any, i: number) => {
+                console.log(result[i])
```
remove unnecessary comments

OmerMunkMoveo replied:
Done

GuyLeiferMoveo commented on src/Components/Restaurants/Restaurants.tsx:
```diff
@@ -80,11 +81,27 @@ const Restaurants = (props: any) => {
+            const result = res.data.data
             const rendered = Object.keys(result).map((keyName: any, i: any) => {
-                if (result[keyName].popular === "true") {
+                if (result[keyName].popular) {
```
the variable is boolean for sure?
because if it is "false" it will be truth condition

OmerMunkMoveo replied:
Fixed

GuyLeiferMoveo commented on src/Components/UI/Input/Input.tsx:
```diff
@@ -1,21 +1,88 @@
+        let value = event.target.value
+        axios.get(`${serverURL}/search/all/${value}`).then((res: any) => {
+            setResults(res.data.data)
+            console.log(res.data.data)
```
remove unnecessary logs

OmerMunkMoveo replied:
Done

GuyLeiferMoveo commented on src/Components/UI/Input/Input.tsx:
```diff
@@ -25,8 +92,24 @@ const Input = (props: any) => {
+                    <ul>
+                        {renderedResults && renderedResults.map(((item: any, index: number) =>
+                                <NavLink className='link' to={`/${item.type}/${item.id}`}>
+                                    <li key={index}>{item.display}</li>
```
if there is an option, give the key non index value. In this case you can give it the value of the ID.

OmerMunkMoveo replied:
Fixed

OmerMunkMoveo committed:
Updated chef of the week section according to the DB
next working on:
1. on click of search item: display the item.
2. styling the dropdown
3. debounce on the search
4. loaders
5. toastifiers

OmerMunkMoveo committed:
Fixed comments for Guy Leifer (reviewer)
